### PR TITLE
 refactor(MessageReaction): ReactionUserStore

### PR DESCRIPTION
**Pull request by Lewdcario**
**Please describe the changes this PR makes and why it should be merged:**
since we're doing OOP stuff MessageReaction.fetchUsers() wasn't really consistent, so I made a store for it, so we can do MessageReaction.users.fetch() now.

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/stores/MessageReactionUserStore.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * A data store to store User models who reacted to a MessageReaction.
+ * @extends {DataStore}
+ */
+class MessageReactionUserStore extends DataStore {
+  constructor(client, iterable, data) {
+    if (!User) User = require('../structures/User');
+    super(client, iterable, User);
+    this.data = data;
```
Comment (bdistin) [inline src/stores/MessageReactionUserStore.js]: This is a reaction, and not something like a data packet ect. Can this be renamed to this.reaction.

File: src/stores/MessageReactionUserStore.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+   * Fetches all the users that gave this reaction. Resolves with a collection of users, mapped by their IDs.
+   * @param {Object} [options] Options for fetching the users
+   * @param {number} [options.limit=100] The maximum amount of users to fetch, defaults to 100
+   * @param {Snowflake} [options.after] Limit fetching users to those with an id greater than the supplied id
+   * @returns {Promise<Collection<Snowflake, User>>}
+   */
+  async fetch({ limit = 100, after } = {}) {
+    const message = this.data.message;
+    const reaction = message.reactions.get(this.data.emoji.id || this.data.emoji.name);
+    const users = await message.client.api.channels[message.channel.id].messages[message.id]
```
Comment (bdistin) [inline src/stores/MessageReactionUserStore.js]: You have access to client via this.client. No need to use message to get client.

File: src/stores/MessageReactionUserStore.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  /**
+   * Fetches all the users that gave this reaction. Resolves with a collection of users, mapped by their IDs.
+   * @param {Object} [options] Options for fetching the users
+   * @param {number} [options.limit=100] The maximum amount of users to fetch, defaults to 100
+   * @param {Snowflake} [options.after] Limit fetching users to those with an id greater than the supplied id
+   * @returns {Promise<Collection<Snowflake, User>>}
+   */
+  async fetch({ limit = 100, after } = {}) {
+    const message = this.data.message;
+    const reaction = message.reactions.get(this.data.emoji.id || this.data.emoji.name);
```
Comment (bdistin) [inline src/stores/MessageReactionUserStore.js]: You already have this, via this.data; which is partially why the comment above to rename to this.reaction.

File: src/stores/MessageReactionUserStore.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    const users = await message.client.api.channels[message.channel.id].messages[message.id]
+      .reactions[this.data.emoji.identifier]
+      .get({ query: { limit, after } });
+    for (const rawUser of users) {
+      const user = message.client.users.create(rawUser);
```
Comment (bdistin) [inline src/stores/MessageReactionUserStore.js]: Again you have client via this.client.

**Comment by devsnek** [comment]
can we possibly do something to define like a `fetchable store` where we provide an endpoint template and it does the rest

**Comment by hydrabolt** [comment]
that seems overkill Gus

**Pull request merged**